src/docfx/lib/log/SourceInfo.cs

```diff
@@ -1,42 +1,53 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-using System.Diagnostics;
-
 namespace Microsoft.Docs.Build
 {
-    [DebuggerDisplay("{Value}")]
-    public sealed class SourceInfo<T> : ISourceInfo
+    public class SourceInfo
     {
-        public readonly T Value;
-
+        /// <summary>
+        /// Path to the source file.
+        /// </summary>
         public readonly string File;
 
-        public readonly Range Range;
+        /// <summary>
+        /// A one based start line value.
+        /// </summary>
+        public readonly int Line;
 
-        object ISourceInfo.Value => Value;
+        /// <summary>
```

Comment:
since "endLine" and "endColumn" are not used anyway, just delete them?

Reply:
https://github.com/Microsoft/vscode/blob/master/src/vs/editor/common/core/range.ts
Even vscode define `range` with `endLine` and `endColumn`
We only have them from YamlNode https://github.com/dotnet/docfx/blob/575c0ad2b242ac79947270f68cedd3e057ae33b0/src/docfx/lib/json/YamlUtility.cs#L178
And we are not exposing them anyway, so they can be removed? @yufeih

Reply:
We have end line|column info for markdown nodes as well. Can add `EndLine, EndColumn` to `Error` to expose it.

Reply:
that;s what i mean, seems current `EndLine and EndColumn` are removed from `Error`

Reply:
added end line|column back to `Error.cs`